book/07-extending-learners.qmd

```diff
@@ -448,25 +541,24 @@ Some learners have a high number of parameters and it is easy to miss out on som
 In addition, if the maintainer of the upstream package changes something with respect to the arguments of the algorithm, the learner is in danger to break.
 Also, new arguments could be added upstream and manually checking for new additions all the time is tedious.
 
-Therefore we have written a "Parameter Check" that runs for every learner asynchronously to the R CMD Check of the package itself. This "Parameter Check" compares the parameters of the `r mlr3book::mlr_pkg("mlr3")` ParamSet against all arguments available in the upstream function that is called during `$train()` and `$predict()`. Again the file is automatically created and opened by `r ref("create_learner()")`, this will be named like `test_paramtest_package_type_key.R`, so in our example `test_paramtest_rpart_classif_rpart.R`.
+Therefore we have written a "Parameter Check" that runs regularly for every learner. This "Parameter Check" compares the parameters of the `r mlr3book::mlr_pkg("mlr3")` ParamSet against all arguments available in the upstream function that is called during `$train()` and `$predict()`. Again the file is automatically created and opened by `r ref("create_learner()")`, this will be named like `test_paramtest_package_type_key.R`, so in our example `test_paramtest_rpart_classif_rpart.R`.
```

Comment:
```suggestion
Therefore we have written a "Parameter Check" that runs regularly for every learner. This "Parameter Check" compares the parameters of the `r mlr3book::mlr_pkg("mlr3")` ParamSet against all arguments available in the upstream function that is called during `$train()` and `$predict()`.
Again the file is automatically created and opened by `r ref("create_learner()")`, this will be named like `test_paramtest_package_type_key.R`, so in our example `test_paramtest_rpart_classif_rpart.R`.
```

I am reading this the third time now - should we reduce redundancies?

Reply:
I think it's fine

Reply:
The Q was rethorical, I think we should not repeat this phrase so often here as it gives a bit the feeling of a partially c/p article content.

Reply:
But it's not a blocker for me, your choice. Just wanted to clarifuy my intentions :)